diff --git a/src/main/java/me/hexedhero/pp/shaded/paperlib/features/chunkisgenerated/ChunkIsGeneratedApiExists.java b/src/main/java/me/hexedhero/pp/shaded/paperlib/features/chunkisgenerated/ChunkIsGeneratedApiExists.java
deleted file mode 100644
index d2f1dcc..0000000
--- a/src/main/java/me/hexedhero/pp/shaded/paperlib/features/chunkisgenerated/ChunkIsGeneratedApiExists.java
+++ /dev/null
@@ -1,10 +0,0 @@
-package me.hexedhero.pp.shaded.paperlib.features.chunkisgenerated;
-
-import org.bukkit.World;
-
-public class ChunkIsGeneratedApiExists implements ChunkIsGenerated {
-   @Override
-   public boolean isChunkGenerated(World var1, int var2, int var3) {
-      return var1.isChunkGenerated(var2, var3);
-   }
-}
