diff --git a/src/main/java/me/hexedhero/pp/shaded/paperlib/features/bedspawnlocation/BedSpawnLocationSync.java b/src/main/java/me/hexedhero/pp/shaded/paperlib/features/bedspawnlocation/BedSpawnLocationSync.java
deleted file mode 100644
index c554e34..0000000
--- a/src/main/java/me/hexedhero/pp/shaded/paperlib/features/bedspawnlocation/BedSpawnLocationSync.java
+++ /dev/null
@@ -1,12 +0,0 @@
-package me.hexedhero.pp.shaded.paperlib.features.bedspawnlocation;
-
-import java.util.concurrent.CompletableFuture;
-import org.bukkit.Location;
-import org.bukkit.entity.Player;
-
-public class BedSpawnLocationSync implements BedSpawnLocation {
-   @Override
-   public CompletableFuture<Location> getBedSpawnLocationAsync(Player var1, boolean var2) {
-      return CompletableFuture.completedFuture(var1.getBedSpawnLocation());
-   }
-}
