diff --git a/src/main/java/me/hexedhero/pp/shaded/paperlib/features/blockstatesnapshot/BlockStateSnapshot.java b/src/main/java/me/hexedhero/pp/shaded/paperlib/features/blockstatesnapshot/BlockStateSnapshot.java
deleted file mode 100644
index e17829f..0000000
--- a/src/main/java/me/hexedhero/pp/shaded/paperlib/features/blockstatesnapshot/BlockStateSnapshot.java
+++ /dev/null
@@ -1,7 +0,0 @@
-package me.hexedhero.pp.shaded.paperlib.features.blockstatesnapshot;
-
-import org.bukkit.block.Block;
-
-public interface BlockStateSnapshot {
-   BlockStateSnapshotResult getBlockState(Block var1, boolean var2);
-}
