diff --git a/src/main/java/me/hexedhero/pp/shaded/paperlib/features/blockstatesnapshot/BlockStateSnapshotNoOption.java b/src/main/java/me/hexedhero/pp/shaded/paperlib/features/blockstatesnapshot/BlockStateSnapshotNoOption.java
deleted file mode 100644
index a646f16..0000000
--- a/src/main/java/me/hexedhero/pp/shaded/paperlib/features/blockstatesnapshot/BlockStateSnapshotNoOption.java
+++ /dev/null
@@ -1,10 +0,0 @@
-package me.hexedhero.pp.shaded.paperlib.features.blockstatesnapshot;
-
-import org.bukkit.block.Block;
-
-public class BlockStateSnapshotNoOption implements BlockStateSnapshot {
-   @Override
-   public BlockStateSnapshotResult getBlockState(Block var1, boolean var2) {
-      return new BlockStateSnapshotResult(true, var1.getState());
-   }
-}
