diff --git a/src/main/java/me/hexedhero/pp/shaded/paperlib/features/inventoryholdersnapshot/InventoryHolderSnapshot.java b/src/main/java/me/hexedhero/pp/shaded/paperlib/features/inventoryholdersnapshot/InventoryHolderSnapshot.java
deleted file mode 100644
index 9f73e3b..0000000
--- a/src/main/java/me/hexedhero/pp/shaded/paperlib/features/inventoryholdersnapshot/InventoryHolderSnapshot.java
+++ /dev/null
@@ -1,7 +0,0 @@
-package me.hexedhero.pp.shaded.paperlib.features.inventoryholdersnapshot;
-
-import org.bukkit.inventory.Inventory;
-
-public interface InventoryHolderSnapshot {
-   InventoryHolderSnapshotResult getHolder(Inventory var1, boolean var2);
-}
